packages/outstatic/src/pages/edit-document/index.tsx

```diff
@@ -70,33 +74,130 @@ export default function EditDocument({ collection }: EditDocumentProps) {
     skip: slug === 'new' || !slug
   })
 
+  const { data: metadata } = useDocumentQuery({
+    variables: {
+      owner: repoOwner || session?.user?.login || '',
+      name: repoSlug,
+      filePath: `${repoBranch}:${
+        monorepoPath ? monorepoPath + '/' : ''
+      }${contentPath}/metadata.json`
+    },
+    fetchPolicy: 'network-only'
+  })
+
   const onSubmit = async (data: Document) => {
     setLoading(true)
     try {
       const document = methods.getValues()
-      const content = mergeMdMeta({ ...data })
+      let content = mergeMdMeta({ ...data })
+      const { data: matterData } = matter(content)
       const oid = await fetchOid()
       const owner = repoOwner || session?.user?.login || ''
       const newSlug = document.slug
 
+      const fullContentPath = `${contentPath}/${collection}/${slug}.md`
```

Comment:
I think we created a bug by using the same path for `metadata.json` and for the `capi`. Basically the path for `metada.json` shouldn't include the monorepo path, but the `capi` should get the full path, otherwise it will save the files at the root level. I've suggested some other code changes to  this file to make it work.

```suggestion
```